lib/setup.c

```diff
@@ -921,6 +926,40 @@ ___SIZE_TS requested_words_still;)
 }
 
 
+___EXP_FUNC(___BOOL, ___fdset_resize)
+   ___P((int fd1,
+         int fd2),
+        (fd1,
+         fd2)
+int fd1;
+int fd2;)
+{
+#ifdef USE_select_or_poll
+
+  ___processor_state ___ps = ___PSTATE;
+  int fd = (fd2 > fd1) ? fd2 : fd1;
+
+  if (fd < ___ps->os.fdset.size)
+    return 0;
+
+  ___fdset_resize_heap_overflow_clear ();
+
+  ___sync_op_struct sop;
+
+  sop.op = OP_FDSET_RESIZE;
+  sop.arg[0] = fd;
+
+  on_all_processors (___PSP &sop);
+
+  return ___fdset_resize_heap_overflow ();
```

Comment:
Yes, i've been thinking about this too. I think we need a second atomic flag for resize in progress, in which case we punt in clearing overflow and merge the results of the allocation.
So basically if either of the concurrent resizes fails with overflow, both will fail. 
This is also compatible with keeping the size in vmstate, as it only grows when there are no failures.